### PR TITLE
Ehsan requests

### PR DESCRIPTION
**Pull request by shanaHirani**

### CODE REVIEW

File: app/src/main/java/com/jetbrains/handson/mpp/ehsan/Test.java [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package com.jetbrains.handson.mpp.ehsan;
+
+public class Test {
```
Comment (ehsanheidari67) [inline app/src/main/java/com/jetbrains/handson/mpp/ehsan/Test.java:3]: While doing a commit you should have a look at all the changes first and then do the commit

File: app/src/main/java/com/jetbrains/handson/mpp/ehsan/data/model/NewsApi.kt [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    val urlToImage: String?,
+)
+
+data class Source(
+    val id: String?,
+    val name: String?,
+)
+
+fun NewsApi.asDomainModel(): News{
+    // if we con not parse the date default value is now
```
Comment (ehsanheidari67) [inline app/src/main/java/com/jetbrains/handson/mpp/ehsan/data/model/NewsApi.kt:45]: It's not a proper logic. In this case all the invalid data would be shown to the user on top.
I'd rather filtering this invalid data out from the list

File: app/src/main/java/com/jetbrains/handson/mpp/ehsan/data/model/NewsApi.kt [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        content = this.content,
+        description = this.description,
+        publishedAt = publishedDate,
+        title = this.title,
+        url = this.url,
+        urlToImage = this.urlToImage
+    )
+}
+
+fun List<NewsApi>.asDomainModel(): List<News> {
```
Comment (ehsanheidari67) [inline app/src/main/java/com/jetbrains/handson/mpp/ehsan/data/model/NewsApi.kt:64]: I wouldn't write such a function on lists. You can map lists in-place wherever you need

File: app/src/main/java/com/jetbrains/handson/mpp/ehsan/data/repository/NewsRepository.kt [hunk 4]
```diff
@@ -66,10 +64,9 @@ class NewsRepository @Inject constructor(
             })
     }
 
-    @RequiresApi(Build.VERSION_CODES.O)
     private fun getTwoDayAgo(): String {
         return LocalDate.now()
-            .minus(Period.of(0, 0, 11))
+            .minus(Period.of(0, 0, 20))
             .format(DateTimeFormatter.ofPattern("dd-MM-yyyy"))
```
Comment (ehsanheidari67) [inline app/src/main/java/com/jetbrains/handson/mpp/ehsan/data/repository/NewsRepository.kt:70]: I would use extension functions to format dates.

File: app/build.gradle [hunk 5]
```diff
@@ -14,11 +14,12 @@ android {
...
     defaultConfig {
         applicationId "com.jetbrains.handson.mpp.ehsan"
-        minSdkVersion 26
+        minSdkVersion 21
         targetSdkVersion 30
         versionCode 1
         versionName "1.0"
-
+        //desugaring Required when setting minSdkVersion to 20 or lower
+        //multiDexEnabled = true
```
[the remark this answers is not in the document]
Comment (shanaHirani) [reply app/build.gradle:22]: done

File: app/src/main/java/com/jetbrains/handson/mpp/ehsan/data/model/NewsApi.kt (hunk 2 above)
Comment (shanaHirani) [reply app/src/main/java/com/jetbrains/handson/mpp/ehsan/data/model/NewsApi.kt:45]: done

File: app/src/main/java/com/jetbrains/handson/mpp/ehsan/data/model/NewsApi.kt (hunk 3 above)
Comment (shanaHirani) [reply app/src/main/java/com/jetbrains/handson/mpp/ehsan/data/model/NewsApi.kt:64]: i also check validation here

File: app/src/main/java/com/jetbrains/handson/mpp/ehsan/Test.java (hunk 1 above)
Comment (shanaHirani) [reply app/src/main/java/com/jetbrains/handson/mpp/ehsan/Test.java:3]: ok

File: app/src/main/java/com/jetbrains/handson/mpp/ehsan/data/repository/NewsRepository.kt (hunk 4 above)
Comment (shanaHirani) [reply app/src/main/java/com/jetbrains/handson/mpp/ehsan/data/repository/NewsRepository.kt]: done

**Review by ehsanheidari67 — APPROVED** [review]

**Pull request merged**